lib/s3routes/routesUtils.ts

```diff
@@ -22,20 +22,33 @@ export type CallApiMethod = (
     callback: (err: ArsenalError | Error | null, ...data: any[]) => void,
 ) => void;
 
+type APIError = {
+    // Some VaultClient errors have the type of error in the "code" property
+    code?: string | number;
+    // Other external API errors may have the type of error in "message" (this is the case
+    // for Metadata API errors for example)
+    message?: string;
+    description?: string;
+}
+
 /**
  * Convert an error to an ArsenalError instance
  * @param err - Error to convert
  * @returns ArsenalError instance
  */
-export function toArsenalError(err: ArsenalError | Error): ArsenalError {
+export function toArsenalError(err: ArsenalError | APIError ): ArsenalError {
```

Comment:
is this `code` specific to VaultClient?
in that case, may be better to type it explicitly like this:
```suggestion
export function toArsenalError(err: ArsenalError | VaultClientError | Error): ArsenalError {
```
or (if it is called either with Arsenal or VaultClient errors)
```suggestion
export function toArsenalError(err: ArsenalError | VaultClientError): ArsenalError {
```

Reply:
> is this `code` specific to VaultClient?

Yes it is specific to it, i made a separate type only for those errors